api/src/main/java/io/minio/MinioClient.java

```diff
@@ -947,55 +977,77 @@ private HttpUrl buildUrl(
       Multimap<String, String> queryParamMap)
       throws IllegalArgumentException, InvalidBucketNameException, NoSuchAlgorithmException {
     if (bucketName == null && objectName != null) {
-      throw new InvalidBucketNameException(
-          NULL_STRING, "null bucket name for object '" + objectName + "'");
+      throw new IllegalArgumentException("null bucket name for object '" + objectName + "'");
```

Comment:
- This method (`buildUrl`) seems too big. It might be a good idea to refactor it and make it smaller.
- For the AWS specific logic in this method, maybe introduce a new class called `AwsUrlBuilder` and use it from here?

Reply:
I think this can be done when we bring builder pattern to constructor